src/app/app.component.ts

```diff
@@ -21,6 +23,10 @@ import { CommonModule } from '@angular/common';
 })
 
 export class AppComponent {
-  constructor() { }
   title = 'Task Manager';
+  constructor(private tasksService:TasksService) { }
+
+  async showAllTasks(){
```

Comment:
same here instead what you could do here is fetch all tasks, subscribe then set... actually I see what you're doing, it works , but I'm not too familiar with this type of approach, the standard would be as I described in above comments